src/live_restore/live_restore_server.c

```diff
@@ -0,0 +1,265 @@
+     * deleted concurrently? It may be easier to have two separate queues.
+     */
+    /*
+     * Open a cursor so no one can get exclusive access on the object. This prevents concurrent
+     * schema operations like drop and rename.
+     */
+    WT_RET(wt_session->open_cursor(wt_session, work_item->uri, NULL, NULL, &cursor));
+
+    /*
+     * We need to get access to the WiredTiger file handle. Given we've opened the cursor we should
+     * be able to access the WT_FH by first getting to its block manager and then the WT_FH.
+     */
+    WT_ASSERT(session, (CUR2BT(cursor)->bm)->is_multi_handle == false);
+    /* This is a bit more than a little gross. */
+    WT_FILE_HANDLE *fh = CUR2BT(cursor)->bm->block->fh->handle;
+    /*
+     * Call the fill holes function. Right now no other reads or writes can be occurring
+     * concurrently.
+     */
+    __wt_verbose_debug2(
+      session, WT_VERB_FILEOPS, "Live restore worker filling holes for: %s", work_item->uri);
+
+    ret = __wti_live_restore_fs_fill_holes(fh, wt_session);
+    WT_TRET(cursor->close(cursor));
+    return (ret);
+}
+
+/*
+ * __live_restore_populate_queue --
+ *     Populate the live restore work queue. Free all objects on failure.
+ */
+static int
+__live_restore_populate_queue(WT_SESSION_IMPL *session, uint64_t *work_count)
+{
+    WT_DECL_RET;
+    WT_LIVE_RESTORE_SERVER *server = &S2C(session)->live_restore_server;
+
+    TAILQ_INIT(&server->work_queue);
+    /*
+     * Open a metadata cursor to gather the list of objects. The metadata file is built from the
+     * WiredTiger.backup file, during __wt_turtle_init. Thus this function must be run after that
+     * function. I don't know if we have a way of enforcing that.
+     */
+    WT_CURSOR *cursor;
+    WT_RET(__wt_metadata_cursor(session, &cursor));
+    WT_LIVE_RESTORE_WORK_ITEM *work_item = NULL;
+    __wt_verbose_debug1(session, WT_VERB_FILEOPS, "Initializing the live restore work %s", "queue");
+
+    *work_count = 0;
+    while ((ret = cursor->next(cursor)) == 0) {
+        const char *uri = NULL;
+        WT_ERR(cursor->get_key(cursor, &uri));
+        if (WT_PREFIX_MATCH(uri, "file:")) {
+            __wt_verbose_debug2(
+              session, WT_VERB_FILEOPS, "Adding an item to the work queue %s", uri);
+            WT_ERR(__wt_calloc_one(session, &work_item));
+            WT_ERR(__wt_strdup(session, uri, &work_item->uri));
+            TAILQ_INSERT_HEAD(&server->work_queue, work_item, q);
+            (*work_count)++;
+        }
+    }
+    WT_ERR_NOTFOUND_OK(ret, false);
+
+    if (0) {
+err:
+        /* The queue insert cannot fail so whatever we've put into it is guaranteed to be there. */
+        __live_restore_work_queue_drain(session);
+
+        /* Free the various items. */
+        __wt_free(session, work_item->uri);
+        __wt_free(session, work_item);
+    }
+    return (ret);
+}
+
+/*
+ * __wt_live_restore_server_init --
```

Comment:
I suggest naming this `_create` to go with all the other subsystem thread creation names (see conn_open.c)

Reply:
af280f343